eq-author/cypress/routingTests/routing_e2e_spec.js

```diff
@@ -0,0 +1,305 @@
+import { question, questionnaire, section } from "../builders";
+import {
+  findByLabel,
+  selectOptionByLabel,
+  navigateToPage,
+  testId
+} from "../utils";
+
+const navigateToRoutingTab = () =>
+  cy
+    .log("Navigating to routing tab")
+    .get(testId("tabs-nav"))
+    .within(() => cy.contains("Routing").click())
+    .url()
+    .should("contain", "routing");
+
+const setRoutingCondition = (questionTitle, label) => {
+  findByLabel("IF").within(() => selectOptionByLabel(questionTitle));
+  cy.get(testId("options-selector")).within(() => cy.contains(label).click());
+};
+
+describe("Routing_e2e", () => {
+  after(() => {
+    cy.visit("/");
+    cy.login();
+    cy.deleteQuestionnaire("Routing e2e");
+  });
+
+  it("builds the questionnaire", () => {
+    cy.visit("/");
+    cy.login();
+    questionnaire.add({ title: "Routing e2e" });
+
+    section.updateInitial({
+      title: "Section 1"
+    });
+
+    question.updateInitial({
+      sectionDisplayName: "Section 1",
+      title: "Section 1, Page 1",
+      answer: {
+        type: "Radio",
+        options: [
+          { label: "Goto S1P4" },
+          { label: "Goto end of questionnaire" }
+        ]
+      }
+    });
+
+    question.add({
+      sectionDisplayName: "Section 1",
+      previousPageDisplayName: "Section 1, Page 1",
+      title: "Section 1, Page 2",
+      confirmation: {
+        title: "Confirmation question for Section 1, Page 2",
+        positive: "Yes",
+        negative: "No"
+      },
+      answer: {
+        type: "Number",
+        label: "Less than 5 goes to S1P4 Greater than 5 goes to EoQ"
+      }
+    });
+
+    question.add({
+      sectionDisplayName: "Section 1",
+      previousPageDisplayName: "Section 1, Page 2",
+      title: "Section 1, Page 3",
+      answer: {
+        type: "Number",
+        label: "Placeholder Answer"
+      }
+    });
+
+    question.add({
+      sectionDisplayName: "Section 1",
+      previousPageDisplayName: "Section 1, Page 3",
+      title: "Section 1, Page 4",
+      answer: {
+        type: "Number",
+        label: "Placeholder Answer"
+      }
+    });
+
+    section.add({ title: "Section 2" });
+
+    question.updateInitial({
+      sectionDisplayName: "Section 2",
+      title: "Section 2, Page 1",
+      answer: {
+        type: "Number",
+        label: "Placeholder Answer"
+      }
+    });
+  });
+
+  it("adds radio routing", () => {
+    navigateToPage("Section 1, Page 1");
+    navigateToRoutingTab();
+    cy.get(testId("btn-add-rule")).click();
+
+    cy.get(testId("routing-editor")).within(() => {
```

Comment:
Some of these tests are shared with `routing_spec.js`.  Can we share them?

Reply:
I didn't really want to share anything here except for the utils as we wanted these tests to be very specific and isolated from the other routing implementation.

Reply:
We could move the actions to builders? So we can reuse the same functionality in both places